openpifpaf/datasets/factory.py

```diff
@@ -1,17 +1,58 @@
 import torch
 
 from .coco import Coco
+from .visdrone import VisDrone
+from .uavdt import UAVDT
+from .eurocity import EuroCity
+from .nightowls import NightOwls
+from .visual_relationship import VisualRelationship
 from .collate import collate_images_targets_meta
-from .constants import COCO_KEYPOINTS, HFLIP
+from .constants import COCO_KEYPOINTS, HFLIP, BBOX_KEYPOINTS, BBOX_HFLIP
 from .. import transforms
 
-COCOKP_ANNOTATIONS_TRAIN = 'data-mscoco/annotations/person_keypoints_train2017.json'
-COCOKP_ANNOTATIONS_VAL = 'data-mscoco/annotations/person_keypoints_val2017.json'
-COCODET_ANNOTATIONS_TRAIN = 'data-mscoco/annotations/instances_train2017.json'
-COCODET_ANNOTATIONS_VAL = 'data-mscoco/annotations/instances_val2017.json'
-COCO_IMAGE_DIR_TRAIN = 'data-mscoco/images/train2017/'
-COCO_IMAGE_DIR_VAL = 'data-mscoco/images/val2017/'
-
+COCOKP_ANNOTATIONS_TRAIN = 'data/mscoco/annotations/person_keypoints_train2017.json'
+COCOKP_ANNOTATIONS_VAL = 'data/mscoco/annotations/person_keypoints_val2017.json'
+COCODET_ANNOTATIONS_TRAIN = 'data/mscoco/annotations/instances_train2017.json'
+COCODET_ANNOTATIONS_VAL = 'data/mscoco/annotations/instances_val2017.json'
+COCO_IMAGE_DIR_TRAIN = 'data/mscoco/images/train2017/'
+COCO_IMAGE_DIR_VAL = 'data/mscoco/images/val2017/'
+
+UAVDT_ANNOTATIONS_TRAIN = 'data/UAV-benchmark-M/GT/'
+UAVDT_ANNOTATIONS_VAL = 'data/UAV-benchmark-M/GT/'
+UAVDT_IMAGE_DIR_TRAIN = 'data/UAV-benchmark-M/train/'
+UAVDT_IMAGE_DIR_VAL = 'data/UAV-benchmark-M/test/'
+
+VISDRONE_ANNOTATIONS_TRAIN = 'data/VisDrone2019/VisDrone2019-DET-train/annotations'
+VISDRONE_ANNOTATIONS_VAL = 'data/VisDrone2019/VisDrone2019-DET-val/annotations'
+VISDRONE_IMAGE_DIR_TRAIN = 'data/VisDrone2019/VisDrone2019-DET-train/images'
+VISDRONE_IMAGE_DIR_VAL = 'data/VisDrone2019/VisDrone2019-DET-val/images'
+
+EUROCITY_ANNOTATIONS_TRAIN = 'data/ECP/{}/labels/train'
+EUROCITY_ANNOTATIONS_VAL = 'data/ECP/{}/labels/val'
+EUROCITY_IMAGE_DIR_TRAIN = 'data/ECP/{}/img/train'
+EUROCITY_IMAGE_DIR_VAL = 'data/ECP/{}/img/val'
+
+NIGHTOWLS_ANNOTATIONS_TRAIN = 'data/nightowls/nightowls_validation.json'
+NIGHTOWLS_ANNOTATIONS_VAL = 'data/nightowls/nightowls_validation.json'
+NIGHTOWLS_IMAGE_DIR_TRAIN = 'data/nightowls/nightowls_validation/'
+NIGHTOWLS_IMAGE_DIR_VAL = 'data/nightowls/nightowls_validation/'
+
+VISUAL_ANNOTATIONS_TRAIN = 'data/visual_relationship/annotations_train.json'
+VISUAL_ANNOTATIONS_VAL = 'data/visual_relationship/annotations_test.json'
+VISUAL_IMAGE_DIR_TRAIN = 'data/visual_relationship/sg_dataset/sg_train_images'
+VISUAL_IMAGE_DIR_VAL = 'data/visual_relationship/sg_dataset/sg_test_images'
+
+dataset_list = {
```

Comment:
No, The visual relationship dataset is not necessary. Since it contains detection labels and can be trained with only cifdet, I kept it here but it can be removed.